lib/src/native/realm_core.dart

```diff
@@ -854,7 +863,13 @@ extension on Pointer<realm_value_t> {
       case realm_value_type.RLM_TYPE_DECIMAL128:
         throw Exception("Not implemented");
       case realm_value_type.RLM_TYPE_OBJECT_ID:
-        throw Exception("Not implemented");
+        final nativeBytes = ref.values.object_id.bytes;
+        final bytes = List<int>.filled(12, 0);
+        for (int i = 0; i < 12; i++) {
+          bytes[i] = nativeBytes[i];
+        }
+
+        return ObjectId.fromBytes(bytes);
```

Comment:
```suggestion
        return ObjectId.fromBytes(cast<Uint8>().asTypedList(12));
```
The `ObjectId.fromBytes` ctor already copies the bytes. Given the memory layout of the `realm_value_t` we can just read the first 12 bytes.